drkafka/src/main/java/com/pinterest/doctorkafka/modules/action/KafkaReportOperation.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.commons.configuration2.AbstractConfiguration;
+import org.apache.commons.configuration2.Configuration;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Properties;
+import java.util.concurrent.Future;
+
+public class KafkaReportOperation extends Action {
+  private static final Logger LOG = LogManager.getLogger(KafkaReportOperation.class);
+  private static final int MAX_RETRIES = 5;
+  private static final EncoderFactory avroEncoderFactory = EncoderFactory.get();
+  private static final SpecificDatumWriter<OperatorAction> avroWriter
+      = new SpecificDatumWriter<>(OperatorAction.SCHEMA$);
+
+  private final static String CONFIG_TOPIC_KEY = "topic";
+  private final static String CONFIG_ZKURL_KEY = "zkurl";
+  private final static String CONFIG_PRODUCER_CONFIG_KEY = "producer";
+  private final static String CONFIG_SECURITY_PROTOCOL_KEY = CONFIG_PRODUCER_CONFIG_KEY + ".security.protocol";
+
+  private final static SecurityProtocol DEFAULT_SECURITY_PROTOCOL = SecurityProtocol.PLAINTEXT;
+
+  private static String configTopic;
+
+  private final static String EVENT_SUBJECT_KEY = "subject";
+  private final static String EVENT_MESSAGE_KEY = "message";
+
+  private static boolean configured = false;
+  private static Producer<byte[], byte[]> kafkaProducer;
+
+  @Override
+  public synchronized void configure(AbstractConfiguration config) throws
+                                                                   ModuleConfigurationException {
+    super.configure(config);
+    if (!configured) {
+      if(!config.containsKey(CONFIG_TOPIC_KEY)){
+        throw new ModuleConfigurationException("Missing config " + CONFIG_TOPIC_KEY + " in plugin " + KafkaReportOperation.class);
+      }
+      configTopic = config.getString(CONFIG_TOPIC_KEY);
+      kafkaProducer = createReportKafkaProducerFromConfig(config);
+      configured = true;
+    }
+  }
+
+  @Override
+  public Collection<Event> execute(Event event) throws Exception {
+    if(isDryRun()) {
+      LOG.info("Dry run: Action {} triggered by event {}", this.getClass(), event.getName());
+    } else if(event.containsAttribute(EVENT_SUBJECT_KEY) && event.containsAttribute(EVENT_MESSAGE_KEY)){
+      String subject = (String) event.getAttribute(EVENT_SUBJECT_KEY);
+      String message = (String) event.getAttribute(EVENT_MESSAGE_KEY);
+      report(subject, message);
+    }
+    return null;
+  }
+
+  public synchronized void report(String subject, String message) throws Exception {
```

Comment:
Since it's sending events to Kafka does it make sense to use `subject`? Original code uses clustername.

Reply:
This arg is to provide the subject of an action, which although currently only takes the cluster's name from the plugins, is not limited to a cluster.